### PR TITLE
FIT-465 fix critical and major code smells

### PR DESCRIPTION
**Pull request by NagaPradeepGajula-osi**
## JIRA
https://osi-cwds.atlassian.net/browse/FIT-465

## Description
Fix suspicious reference comparison of Boolean values and Reduce the number of conditional operators (7) used in the expression. 

## Tests
- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests
No tests are added was only fixing sonarqube changes

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [ ] I ran all the linters and static analyzers for the project (SonarQube, eslint, rubocop, etc).
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.

### CODE REVIEW

**Comment by NagaPradeepGajula-osi** [comment]
retest this please

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java [hunk 1]
```diff
@@ -120,9 +120,9 @@ private boolean areDomainsAndItemsValid(
     AssessmentJson assessmentJson = assessment.getState();
     boolean result = true;
     for (DomainJson domain : assessmentJson.getDomains()) {
-      result = result & validateCaregiverDomains(context, domain);
+      result = result && validateCaregiverDomains(context, domain);
```
Comment (wmitchell-cambria) [inline src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: Make sure @denys-davydov looks at these. I believe these were intentional at the time, as the `validateCaregiverDomains` has a side effect of generating all of the validation errors at once. Short-circuiting is not desired.

Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: added. thanks for headsup

**Comment by NagaPradeepGajula-osi** [comment]
retest this please

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java (hunk 1 above)
Comment (LeonidMarushevskyi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: In this case & is needed. We don't want to have short-circuit operation. Will is correct.

Comment (LeonidMarushevskyi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: Just suppress this warning for this case.

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java [hunk 2]
```diff
@@ -120,9 +120,9 @@ private boolean areDomainsAndItemsValid(
...
       for (ItemJson item : domain.getItems()) {
-        result = result & validateItemRatings(context, assessmentJson, domain, item);
+        result = result && validateItemRatings(context, assessmentJson, domain, item);
```
Comment (LeonidMarushevskyi) [inline src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: The same here. Should be &.

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java (hunk 1 above)
Comment (LeonidMarushevskyi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: Should be &. It is intentional. Will described correct motivation above.

Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: Sure. I will suppress it.

Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: Sure. I will suppress it.

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java (hunk 2 above)
Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: 👍

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java (hunk 1 above)
Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: this is done can you re review

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java [hunk 3]
```diff
@@ -134,22 +135,24 @@ private boolean validateItemRatings(
       DomainJson domain,
       ItemJson item) {
     final boolean isAssessmentUnderSix = isTrue(assessmentJson.getUnderSix());
-    if ((isAssessmentUnderSix
+    final boolean underSixAssessment = isAssessmentUnderSix
```
Comment (denys-davydov) [inline src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: `underSixAssessment` variable name is confusing. `isEmptyRequiredUnderSixItem` will reflect the nature of the variable better.

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java [hunk 4]
```diff
@@ -134,22 +135,24 @@ private boolean validateItemRatings(
...
             && isTrue(domain.getUnderSix())
             && isNotBlank(item.getUnderSixId())
-            && item.getRating() == -1)
-        || (!isAssessmentUnderSix
+            && item.getRating() == -1;
+    final boolean overSixAssessment = !isAssessmentUnderSix
```
Comment (denys-davydov) [inline src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: `overSixAssessment ` variable name is confusing. `isEmptyRequiredAboveSixItem` will reflect the nature of the variable better.

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java (hunk 3 above)
Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: done

File: src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java (hunk 4 above)
Comment (NagaPradeepGajula-osi) [reply src/main/java/gov/ca/cwds/cans/validation/ValidAssessmentValidator.java]: done

**Pull request merged**